cmd/thanos/query_frontend.go

```diff
@@ -217,6 +224,19 @@ func runQueryFrontend(
 	cfg *queryFrontendConfig,
 	comp component.Component,
 ) error {
+	if len(cfg.orgIdHeaders) != 0 && cfg.TenantHeader != "" {
```

Comment:
Actually, won't this always evaluate true when setting the `orgIdHeader`, as the `TenantHeader` has a non-empty default value?

Reply:
Hummmm, true. I'm starting to think it will be much simpler to output a warning and give priority to the tenant header configuration. 🤔 

Let me see what I can do!